exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/parser/CompoundIdentifierConverter.java

```diff
@@ -115,6 +119,18 @@ public SqlNode visitChild(
           enableComplex = true;
         }
       }
+      if (expr.getKind() == SqlKind.SELECT) {
+        if (((SqlSelect) expr).getFrom() instanceof DrillCompoundIdentifier) {
+          fullSchemasSet.add((DrillCompoundIdentifier) ((SqlSelect) expr).getFrom());
+        } else if (((SqlSelect) expr).getFrom() instanceof SqlJoin) {
```

Comment:
Seems you are adding the schema-qualified table identifier. Will this logic work for nested subqueries? Normally, the name resolution will happen under certain name scope. I'm not clear that adding all the table identifier to this "fullSchemasSet" will work for name scope based resolution.

Can you try some queries with nested subqueries with different schema-qualified tables?